SonarProjectConfig.xml

```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<SonarProjectConfig xmlns="http://www.sonarsource.com/msbuild/analyzer/2021/1">
+  <AnalysisConfigPath>c:\Projects\1h with Hacker\AwesomeBankWeb\.sonarqube\conf\SonarQubeAnalysisConfig.xml</AnalysisConfigPath>
+  <ProjectPath>C:\Projects\1h with Hacker\AwesomeBankWeb\AwesomeBankWeb.CSharp\AwesomeBankWeb.CSharp.csproj</ProjectPath>
+  <OutputDir>C:\Projects\1h with Hacker\AwesomeBankWeb\.sonarqube\out\0</OutputDir>
```

Comment:
Old name: `ProjectOutFolderPath.txt`
Original from `SonarQubeAnalysisConfig.xml`: `SonarOutputDir`

I'm avoiding `Sonar` prefix as everything in this `SonarProjectConfig` is `Sonar`.
I'm avoiding `Project` prefix as everything in this `SonarProjectConfig` is `Project`-scoped with notable exception of `ProjectPath` as it's not a prefix but a descriptor.

Reply:
can you use the same terminology as used in the Analyzer?
Like `WorkDirectoryBasePath` ? (see in [UtilityAnalyzerBase](https://github.com/SonarSource/sonar-dotnet/blob/96ad8b26b3c9bce808b747bbac9d3be07dd6940e/analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs#L71))

Or AnalyzerWorkDir in the Java SQ Plugin code - [AbstractPropertyDefinitions](https://github.com/SonarSource/sonar-dotnet/blob/3135cc639f2024e5e6f8a836e2afbee51cdc204e/sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractPropertyDefinitions.java#L168)

Reply:
I'll be happy to challenge and rename `WorkDirectoryBasePath` on the way, as it's not a good name 😄 

What's actually work directory? The current directory that the Scanner was run from? Current directory that the build was run from? Or is it `project` specific `output` `dir` from `config` AKA `(Sonar)ProjectConfig.OutputDir` 🙂

Reply:
I'm fine as long as we agree on a common vocabulary to use

Output dir is also `<PROJECT>\SastTest2\bin\Debug\netcoreapp2.2`, where the `Sast2.dll.RoslynCA.json` is output, but that's used as `sonar.cs.roslyn.reportFilePath`

Given we use `sonar.cs.analyzer.projectOutPath` - can it be `ProjectOutPath` and use this everywhere - Scanner, Config files, analyzer and SQ plugin?

Reply:
`reportFilePath` has specific and quite accurate meaning.

`projectOutPath` is close enough to discuss. I'm generally avoiding `Project` prefix. It's aligned with other `*Path` fields in the XML. So `OutPath` it is?

Reply:
The problem is that `out` is the name of the folder. `out\0`  is actually the `ProjectOutPath`.

I'm fine with `OutPath` as well, as the config is per-project.